spec/puppetserver/ca/action/list_spec.rb

```diff
@@ -66,5 +67,40 @@
       expect(out.string).to match(/Signed Certificates:.*foo.*\(SHA256\).*three.*alt names:.*"DNS:foo", "DNS:bar".*/m)
       expect(out.string).to match(/Missing Certificates:.*fake.*/m)
     end
+
+    it 'errors when unknown format option is passed in' do
+      allow(action).to receive(:get_all_certs).and_return(result)
+      exit_code = action.run({'certname' => ['foo', 'baz'], 'format' => 'test'})
+      expect(exit_code).to eq(1)
+    end
+
+    it 'does not throw error when json format option is passed in' do
+      parsed_output = nil
+      allow(action).to receive(:get_all_certs).and_return(result)
+      exit_code = action.run({'certname' => ['foo', 'baz'], 'format' => 'json'})
+      expect(exit_code).to eq(0)
+      expect {
+        parsed_output = JSON.parse(out.string)
+      }.not_to raise_error
+      expect(parsed_output).to be_a_kind_of(Hash)
+    end
+
+    it 'does not throw error when text format option is passed in' do
+      allow(action).to receive(:get_all_certs).and_return(result)
+      exit_code = action.run({'certname' => ['foo', 'baz'], 'format' => 'text'})
+      expect(exit_code).to eq(0)
+      expect(out.string).to match(/Signed Certificates:.*foo.*\(SHA256\).*three.*alt names:.*"DNS:foo", "DNS:bar".*/m)
+    end
+
+    it 'returns the correct output, including empty keys with the --all option' do
+      parsed_output = nil
+      allow(action).to receive(:get_all_certs).and_return(result)
+      exit_code = action.run({'all' => true, 'format' => 'json'})
+      expect(exit_code).to eq(0)
+      expect {
+        parsed_output = JSON.parse(out.string)
+      }.not_to raise_error
+      expect(parsed_output).to be_a_kind_of(Hash)
```

Comment:
I think it would be good to test that all the keys are present in this hash.

Reply:
Right on. I'll look into it.